arevindh/AdminLTE
# One-Click Update & Uninstall + Grouped Bar Chart + Many fixes

Aha! I knew I could download more speed.

How it started:

* arevindh/pihole-speedtest#125

How it's going:
* arevindh/pihole-speedtest#130
* arevindh/pi-hole#51
* This PR

Features:
* New bar chart
  * Switch per-browser and
  * Change the default for new browsers right from settings
* Mod the Mod
  * Update and/or [Un,Re]Install directly from settings
  * Failsafe
  * Offline uninstall
* Speedtest
  * Run a speedtest manually with a button in settings

Fixes:

* Chart
  * Set fixed height
  * Revamped colors
  * Re-resensitized tooltip
  * Put axes in their rightful places
  * Fixed mobile display issues
  * Removed legend
  * Hide date when only today
* Settings
  * Improved layout
  * Changed Save to Confirm and colored it red when history is to be cleared
  * Removed script selector as speedtest-cli is no longer supported
  * Properly handled custom server input, or lack thereof
* Speedtest
  * Properly handled failure
  * Read results once

Before:

![image](https://user-images.githubusercontent.com/21136719/204110416-849f2247-103f-4eea-90e0-a631c6b1c076.jpeg)

After:

![image](https://user-images.githubusercontent.com/21136719/212464232-ac20b5ab-5b33-4d10-96f5-3a4299e98c3a.jpeg)

![image](https://user-images.githubusercontent.com/21136719/212546970-99efdeaf-40f8-4e27-9aac-916754b8cbdd.jpeg)

![image](https://user-images.githubusercontent.com/21136719/213914589-727fd191-92ac-465d-a184-94d4eff6b356.jpeg)
